src/main/java/hexgui/htp/AnalyzeDefinition.java

```diff
@@ -0,0 +1,188 @@
+    else m_type = AnalyzeType.NONE;
+    m_label = array[1];
+    m_command = array[2];
+  }
+
+  public AnalyzeDefinition(AnalyzeType type, String label, String command) {
+    m_type = type;
+    m_label = label;
+    m_command = command;
+  }
+
+  public String getCommand() {
+    return m_command;
+  }
+
+  public String getLabel() {
+    return m_label;
+  }
+
+  public AnalyzeType getType() {
+    return m_type;
+  }
+
+  /**
+   * Should the response be shown as text. Returns true for types that should be shown (not
+   * necessarily only) as text to the user. That is string and variation commands.
+   */
+  public boolean isTextType() {
+    return m_type == AnalyzeType.STRING
+        || m_type == AnalyzeType.HSTRING
+        || m_type == AnalyzeType.HPSTRING
+        || m_type == AnalyzeType.PSTRING
+        || m_type == AnalyzeType.VAR
+        || m_type == AnalyzeType.VARC
+        || m_type == AnalyzeType.VARW
+        || m_type == AnalyzeType.VARB
+        || m_type == AnalyzeType.VARP
+        || m_type == AnalyzeType.VARPO;
+  }
+
+  public boolean needsColorArg() {
+    return (m_command.indexOf("%c") >= 0);
+  }
+
+  public boolean needsFileArg() {
+    return (m_command.indexOf("%f") >= 0);
+  }
+
+  public boolean needsFileOpenArg() {
+    return (m_command.indexOf("%r") >= 0);
+  }
+
+  public boolean needsFileSaveArg() {
+    return (m_command.indexOf("%w") >= 0);
+  }
+
+  public boolean needsOnlyPointArg() {
+    return (needsPointArg()
+        && !needsColorArg()
+        && !needsFileArg()
+        && !needsFileOpenArg()
+        && !needsFileSaveArg()
+        && !needsPointListArg()
+        && !needsStringArg()
+        && !needsOptStringArg());
+  }
+
+  public boolean needsOnlyPointAndColorArg() {
+    return (needsPointArg()
+        && needsColorArg()
+        && !needsFileArg()
+        && !needsFileOpenArg()
+        && !needsFileSaveArg()
+        && !needsPointListArg()
+        && !needsStringArg()
+        && !needsOptStringArg());
+  }
+
+  public boolean needsPointArg() {
+    return (m_command.indexOf("%p") >= 0);
+  }
+
+  public boolean needsPointListArg() {
+    return (m_command.indexOf("%P") >= 0 || m_type == AnalyzeType.EPLIST);
+  }
+
+  public boolean needsStringArg() {
+    return (m_command.indexOf("%s") >= 0);
+  }
+
+  public boolean needsOptStringArg() {
+    return (m_command.indexOf("%o") >= 0);
+  }
+
+  public static ArrayList<AnalyzeDefinition> read(String programAnalyzeCommands)
+      throws ErrorMessage {
+    if (programAnalyzeCommands == null) throw new ErrorMessage("no analyze commands!");
+
+    Reader stringReader = new StringReader(programAnalyzeCommands);
+    BufferedReader reader = new BufferedReader(stringReader);
+    return readConfig(reader, "program response to hexgui-analyze_commands", null);
+  }
+
+  private final AnalyzeType m_type;
+
+  private final String m_label;
+
+  private final String m_command;
+
+  private static ArrayList<AnalyzeDefinition> readConfig(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.